sdks/python/src/opik/evaluation/tasks_scorer.py

```diff
@@ -33,8 +33,18 @@ def _score_test_case(
                 score_results += result
             else:
                 score_results.append(result)
-        except exceptions.ScoreMethodMissingArguments:
-            raise
+        except exceptions.ScoreMethodMissingArguments as e:
```

Comment:
I think if ScoreMethodMissingArguments exception was caught - it's critical enough to raise an error, that's why we separated this error from `except Exception as e` and it makes total sense to me. So I think we should revert this change back.

Additionaly, in the updated metrics you are raising `TypeError` now (because backward compatible deprecation with optional arguments is not handled by `arguments_helpers.raise_if_score_arguments_are_missing`), I think it will be better to raise `ScoreMethodMissingArguments`, and it will work as it works for every other metric when the arguments are missing. (I tested both ways and what I'm offering works nicer IMO)

Reply:
Reverted